File: 1138-grumpy-bookstore-owner/grumpy-bookstore-owner.java
class Solution {
    public int maxSatisfied(int[] customers, int[] nums, int k) {
        int n = nums.length;
        
        int start = 0;
        int max = 0;
        int cust = 0;

        for( int i=0; i<n; i++){
            if(nums[i]==0){
                cust += customers[i];
            }
        }

        for(int end = 0; end<n; end++){
            if(nums[end]==1){
                cust += customers[end];
            }
            
            if(end - start +1 > k){
                if(nums[start] == 1){
                    cust -= customers[start];
                }
                start++;
            }
            max = Math.max(cust,max);
        }

        return max;
    }
}

File: 1138-grumpy-bookstore-owner/grumpy-bookstore-owner.ts
function maxSatisfied(customers: number[], grumpy: number[], minutes: number): number {
    let result : number = 0;
    let n = customers.length;

    for(let i=0; i<n; i++) {
        result += ((1- grumpy[i]) * customers[i]);
    }

    let start : number = 0;
    let maxSum = 0;
    let sum = 0;

    for(let end:number = 0; end<n; end++) {
        sum += (grumpy[end]) * customers[end];

        while(end - start + 1 > minutes) {
            sum -= (grumpy[start]) * customers[start];
            start++;
        }

        maxSum = Math.max(sum, maxSum);
    }

    return maxSum + result;
};